### PR TITLE
first pass at adding docstrings

### PR DESCRIPTION
**Pull request by cormullion**
I had a go at adding docstrings to `src/Iterators.jl`, since it's always nice to have some syntax info on hand. I also added a `docs/` directory, which can be used as you migrate to using Documenter.jl (although the entire process is more involved, with the `gh-pages` branch and Travis keys stuff... 

Logo.png is just a placeholder...

### CODE REVIEW

**Comment by ararslan** [comment]
It would be best to propose the logo change at https://github.com/JuliaCollections/Roadmap/issues/2 to get feedback first.

**Comment by cormullion** [comment]
@ararslan oh it's just a placeholder, they can chuck it away and run a competition if they want to!

**Comment by ararslan** [comment]
I still recommend posting it at the roadmap issue so it's more visible. It just might become the org logo. 😉

**Comment by cormullion** [comment]
@iamed2 Cool. I had left the `docs/build/` in for testing, sorry, it's added to `.gitignore` now. I've also done some formatting.

(@ararslan I was thinking "iterators", not "collections". I'll have to ponder anew tomorrow...)

**Comment by ararslan** [comment]
docs/build is added to the gitignore but it still has to be unstaged from the commit. You can do that using `git rm` if I'm not mistaken.

**Comment by cormullion** [comment]
@ararslan thanks! I'm not a git-person... 😀

File: docs/src/functionindex.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Index
+
+```@index
+```
```
Comment (ararslan) [inline docs/src/functionindex.md]: Did you mean for this to be empty?

Comment (cormullion) [inline docs/src/functionindex.md]: I think if it's empty it defaults to doing everything...

Comment (ararslan) [inline docs/src/functionindex.md]: Oh, okay. I'm not so good with Documenter, I just wanted to make sure. 🙂

**Comment by cormullion** [comment]
Needs the new logo... :)

**Comment by iamed2** [comment]
The source branch was removed but this is still mergeable. I'll replace the logo myself and do a manual command-line merge. Hopefully I don't mess things up!

**Comment by iamed2** [comment]
Hmm apparently that is not possible. I'll try merging using the online interface and modifying the logo after.

**Pull request merged**